slurm/slurm.py

```diff
@@ -0,0 +1,108 @@
+import os
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+os.environ["TEST_ENVAR_OUTSIDE"] = 'defined'
+
+affinity_script = """
+import os
+affinity = os.sched_getaffinity(0)
+print("num_cores: " + str(len(affinity)))
+for aff in affinity:
+    print("affinity: " + str(aff))"""
+
+
+class SlurmTestBase(rfm.RunOnlyRegressionTest):
+    descr = "Slurm test: "
+    valid_systems = required
+    valid_prog_environs = required
+    time_limit = '10m'
+    num_tasks = 1
+    num_tasks_per_node = 1
+    num_cpus_per_task = 1
+
+
+@rfm.simple_test
+class SbatchCleanEnvTest(rfm.RunOnlyRegressionTest):
+    descr += "sbatch starts in a clean environment"
+    exe = 'print(os.getenv("TEST_ENVAR_OUTSIDE") is None)'
+    executable = f"python3 -c 'import os;{exe}'"
```

Comment:
why split this in two strings?

Reply:
it looks cleaner to me this way, but i'll merge it if you prefer